Improve docs on private dependencies (#10969)
dagger/dagger: docs/current_docs/extending/modules/module-dependencies.mdx

Installing a module using a local path (relative or absolute) is only possible if your module is within the repository root (for Git repositories) or the directory containing the `dagger.json` file (for all other cases).
:::

### Private modules

> You can also install private modules from remote repositories, as long as you have access to them. Dagger supports authentication via both HTTPS (using Git credential managers) and SSH (using `SSH_AUTH_SOCK`). For more information, see [Remote Repositories](/extending/remote-repositories).

Reviewer: ```suggestion
You can also install private modules from remote repositories, as long as you have access to them. Dagger supports authentication via both HTTPS (using Git credential managers) and SSH (using `SSH_AUTH_SOCK`). For more information, see the [documentation on remote repositories](../extending/modules/remote-repositories.mdx).
```

We use filesystem paths in links because Docusaurus can then detect breaks when building the docs site. It is unable to detect breaks when using URL paths.

> When installing a private module, you can use either the normal ref style (preferred if possible, since it allows any available auth method to be used) or be explicit, and specify the protocol. For example, the following commands install the same private module:

Reviewer: ```suggestion
When installing a private module, you can use either the normal ref style (preferred if possible, since it allows any available authentication method to be used), or explicitly specify the protocol. For example, the following commands all install the same private module:
```
